elastic/elastic-agent
# Kubernetes standalone manifest: remove STATE_PATH, replace initcontainer image

## What does this PR do?

- remove STATE_PATH
- in case hints are used - `inputs.d` with all templates will be stored under `/usr/share/elastic-agent/state`
- replaced `busybox` image with the image used for the main container `docker.elastic.co/beats/elastic-agent` to avoid TLS errors

## Why is it important?

- remove STATE_PATH - to use the default config location - `/usr/share/elastic-agent/state`, in this location `hostPath` is mounted to store the agent state
- replaced `busybox` - to remove extra dependency and use the same image that is already used for the main container. With the busybox:1.28 image I got this error `wget: TLS error from peer (alert code 80): 80`, after updating the image to the latest version (1.36) I still got `wget: note: TLS certificate validation not implemented`, even though it is not the error, but as suggested in this issue - https://github.com/docker-library/busybox/issues/64#issuecomment-510194111 busybox shouldn't be used.

## Checklist




## Author's Checklist



## How to test this PR locally
1. Deploy elastic-agent-standalon with uncommented hints related comments
2. deploy redis pod:
```
apiVersion: v1
kind: Pod
metadata:
  name: redis
  annotations:
    co.elastic.hints/package: redis
    co.elastic.hints/data_streams: info, key
    co.elastic.hints/host: '${kubernetes.pod.ip}:6379'
    co.elastic.hints/info.period: 5s
    co.elastic.hints/key.period: 5s
  labels:
    k8s-app: redis
    app: redis
spec:
  containers:
  - image: redis
    imagePullPolicy: IfNotPresent
    name: redis
    ports:
    - name: redis
      containerPort: 6379
      protocol: TCP
```

3. run to check templates are present:
```
kubectl exec -it elastic-agent-standalone-6bmrm bash
...
Defaulted container "elastic-agent-standalone" out of: elastic-agent-standalone, k8s-templates-downloader (init)
root@kind-control-plane:/usr/share/elastic-agent# ls state/inputs.d/
activemq.yml   checkpoint.yml      ....
```
4. check the `/elastic-agent inspect -v --variables --variables-wait 2s`

[inspect.yaml.txt](https://github.com/elastic/elastic-agent/files/15127899/inspect.yaml.txt)

## Related issues


- Relates https://github.com/elastic/elastic-agent/issues/4611

## Use cases



## Screenshots
redis metrics are ingested:
<img width="2548" alt="Screenshot 2024-04-26 at 09 45 44" src="https://github.com/elastic/elastic-agent/assets/28299531/16528ad5-4b39-4db7-a641-a2d85ecc2366">


## Logs



## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
